projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts

```diff
@@ -1,22 +1,30 @@
 import { Injectable } from '@angular/core';
 import { Actions, Effect, ofType } from '@ngrx/effects';
-import {catchError, exhaustMap, map, switchMap} from 'rxjs/operators';
+import { catchError, exhaustMap, map, switchMap, withLatestFrom } from 'rxjs/operators';
 import { of } from 'rxjs';
 import { BookmarksApiService } from '../../annotations/bookmarks-api.service';
 import * as bookmarksActions from '../actions/bookmarks.action';
+import * as fromStore from '../reducers';
+import { select, Store } from '@ngrx/store';
+import * as fromAnnotations from '../selectors/annotations.selectors';
+import * as fromBookmarks from '../selectors/bookmarks.selectors';
+import uuid from 'uuid';
+
 
 @Injectable()
 export class BookmarksEffects {
 
   constructor(private actions$: Actions,
+              private store: Store<fromStore.AnnotationSetState>,
               private bookmarksApiService: BookmarksApiService) {}
 
   @Effect()
   loadBookmarks$ = this.actions$.pipe(
     ofType(bookmarksActions.LOAD_BOOKMARKS),
-    map((action: bookmarksActions.LoadBookmarks) => action.payload),
-    exhaustMap((url) =>
-      this.bookmarksApiService.getBookmarks(url)
+    withLatestFrom(this.store.pipe(select(fromAnnotations.getDocumentIdSetId))),
+    map(([,docSetId]) => docSetId.documentId),
```

Comment:
is not recommended to be connecting store to effect but rather u should pass payload (document id )in the actions plus you are not unsubscribing which is a memory leak